products.php

```diff
@@ -0,0 +1,51 @@
+<?php
+require_once 'src/Models/CategoryModel.php';
+require_once 'src/Services/ProductsDisplayService.php';
+require_once 'src/Models/ProductModel.php';
+require_once 'src/Factory/furnitureDatabaseConnector.php';
+require_once 'src/Entities/ProductsEntity.php';
+
+$db = furnitureDatabaseConnector::connect();
+define('NUM_OF_CATEGORIES', 11);
+$id = $_GET["id"];
+if (!isset($id) || !is_numeric($id) || $id > NUM_OF_CATEGORIES || $id < 1) {
+    echo '<h1>Invalid id number</h1>
+            <a href="index.php"><button>Home</button></a>';
+    return;
```

Comment:
It would be better if any error message were shown within the header and footer of the Furniture Store site, using the same style and branding as the rest of our site. This could be a very common error for users to encounter (e.g. they followed a link from an old email to a product no longer stocked). Consider how e.g. Amazon deals with this situation.